### PR TITLE
[dev-menu] update dev menu dev settings

### PR DESCRIPTION
**Pull request by ajsmth**
# Why

Follow up to what we discussed today - this updates the dev menu buttons to be disabled (or potentially we just don't render them) when loading a react app that does not support certain functionality - for example loading a EAS update means we can't toggle hot reloading or the remote debugger 

A similar thing can be done for the home button, which can have conditional rendering as well (e.g only when the launcher is available) but I will do this in a separate PR after getting feedback from this one

# How

- first I updated the DevSettings module to use swift instead of ObjC 
- then I added the availability of each dev menu setting (to the best of my ability) and had them show a disabled state - we could just remove these buttons as well but it might look a bit bare

# Test Plan

This is only setup for iOS and is meant to get a bit of feedback before I go ahead and implement this in Android and actually clean up the React styles / markup

will update w/ unit tests to ensure the rendering is as expected when a given service is unavailable

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [x] This diff will work correctly for `expo build` (eg: updated `@expo/xdl`).
- [x] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

**Review by esamelson — COMMENTED** [review]
Feel free to disregard, but one thought I had is that this could be confusing if people don't understand why these actions are disabled sometimes. Maybe it's worth showing an explicit message like "This feature is only available in development mode" either on the screen somewhere or if people try to tap?

**Comment by ajsmth** [comment]
>Feel free to disregard, but one thought I had is that this could be confusing if people don't understand why these actions are disabled sometimes. Maybe it's worth showing an explicit message like "This feature is only available in development mode" either on the screen somewhere or if people try to tap?

I was thinking the same thing!

File: packages/expo-dev-menu/ios/EXDevMenuDevSettings.swift [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+      devSettings["isDebuggingRemotely"] = bridgeSettings.isDebuggingRemotely;
+      devSettings["isElementInspectorShown"] = bridgeSettings.isElementInspectorShown;
+      devSettings["isHotLoadingEnabled"] = bridgeSettings.isHotLoadingEnabled;
+      devSettings["isPerfMonitorShown"] = bridgeSettings.isPerfMonitorShown;
+      devSettings["isRemoteDebuggingAvailable"] = bridgeSettings.isRemoteDebuggingAvailable
+      devSettings["isHotLoadingAvailable"] = bridgeSettings.isHotLoadingAvailable
+      devSettings["isPerfMonitorAvailable"] = isPerfMonitorAvailable
+       
+      // what to do?
+      // devSettings["isElementInspectorAvailable"] = ??
```
[the remark this answers is not in the document]
Comment (ajsmth) [reply packages/expo-dev-menu/ios/EXDevMenuDevSettings.swift:39]: okay great I'll update to use the manifest!

**Comment by ajsmth** [comment]
@esamelson I've updated the dev menu to use EXManifest proper (instead of json / dictionary) which was something we discussed previously, as well as to use the value to determine if element inspector is available as suggested above

**Review by esamelson — APPROVED** [review]
Looks good -- I still think it might be worth showing some brief message or toast if features are disabled, but fine if that's outside the scope of this PR.

**Comment by ajsmth** [comment]
> Looks good -- I still think it might be worth showing some brief message or toast if features are disabled, but fine if that's outside the scope of this PR.

I will create a linear task to follow up on this UI

**Pull request merged**